lib/MySQL_Monitor.cpp

```diff
@@ -1138,6 +1145,7 @@ void * monitor_read_only_thread(void *arg) {
 			free(mmsd->mysql_error_msg);
 			mmsd->mysql_error_msg = new_error;
 			proxy_error("Timeout on read_only check for %s:%d after %lldms. Unable to create a connection. If the server is overload, increase mysql-monitor_connect_timeout. Error: %s.\n", mmsd->hostname, mmsd->port, (now-mmsd->t1)/1000, new_error);
+			MyHGM->p_update_mysql_error_counter(p_mysql_error_type::proxysql, mmsd->hostgroup_id, mmsd->hostname, mmsd->port, ER_PROXYSQL_READ_ONLY_CHECK_CONN_TIMEOUT);
```

Comment:
Counter in `create_new_connection` is incrementing `p_mysql_error_type::mysql` counter, this is incrementing `p_mysql_error_type::proxysql` counter, the first one reflects the error in the backend connection, this one the internal ProxySQL error generated because it was a `READ_ONLY_CHECK_CONN_TIMEOUT`. This second one offers information missing in the first one.

Reply:
Why not using `ER_PROXYSQL_READ_ONLY_CHECK_TIMEOUT` ?
Too many variables seems an over-engineering, especially because it means creating a double number of time series. A timeout happened, this is all we need to know.
Same comment applies to all the others.